Modules/mmapmodule.c

```diff
@@ -682,6 +682,86 @@ mmap__exit__method(PyObject *self, PyObject *args)
     return _PyObject_CallMethodId(self, &PyId_close, NULL);
 }
 
+static PyObject *
+mmap__repr__method(PyObject *self)
+{
+    PyObject *repr;
+    mmap_object *m_obj;
+    m_obj = (mmap_object *) self;
+
+    const char *reprfmt;
+    const char *access_str;
+
+    Py_ssize_t size = m_obj->size;
+
+    switch (m_obj->access) {
+        case ACCESS_DEFAULT:
+            access_str = "ACCESS_DEFAULT";
+            break;
+
+        case ACCESS_READ:
+            access_str = "ACCESS_READ";
+            break;
+
+        case ACCESS_WRITE:
+            access_str = "ACCESS_WRITE";
+            break;
+
+        case ACCESS_COPY:
+            access_str = "ACCESS_COPY";
+            break;
+
+        default:
+            PyErr_SetString(PyExc_IOError, "Unknown access mode for mmap object.");
+            return NULL;
+    }
+
+    const char *tp_name = self->ob_type->tp_name;
+    int fd = m_obj->fd;
+
+    if (m_obj -> data == NULL) {
+
+        reprfmt = "<%s is_closed=True fileno=%d access=%s>";
+        repr = PyUnicode_FromFormat(reprfmt, tp_name, fd, access_str);
+
```

Comment:
No need for the wrapper empty lines. And most C APIs could fail and we need to check it.